### ISSUE TITLE
Questions About Custom Tensor Filter Execution and Data Handling in NNStreamer

### ISSUE THREAD

**Issue by an15376** [nnstreamer/nnstreamer#4669]
I am developing a custom tensor filter for NNStreamer, using the Android Java example as a reference. While I have made progress, there are still several challenges and uncertainties that I would like to address. Below are my questions:

Location of Function Execution Code:
Where in the code are the functions defined in the custom tensor filter actually executed? I would like to know the specific part of the code that triggers these functions so I can control when and how they are executed.

Guidelines for Custom Filter Functions:
When creating a custom tensor filter for a new framework, are there any guidelines or naming conventions for the internal functions? For example, how should functions for data input, inference, and output be defined or structured?

Location of Tensor Data Information:
During the pipeline process, where is the tensor data stored after receiving camera input, applying tensor transformations, and before being fed into the model? Is there a way to log or inspect this data at a specific point in the code?

Verification of Data Type Transformation:
When converting camera input data to tensor format through tensor transform, how can I verify that the data type has been correctly transformed? Are there any tools or logging mechanisms in NNStreamer that can confirm the accuracy of the data type conversion?

Issue with Model Invocation:
While I have successfully loaded the model in the custom filter and retrieved the tensor input/output information, I am unable to execute the invoke function. Could you clarify where and how the invoke function is triggered in the pipeline?

Your guidance on these questions will greatly help me understand the execution process and resolve the issues I’m currently facing. Thank you for your support!

**Comment by myungjoo** [comment]
> Guidelines for Custom Filter Functions:

If you are applying a new framework that may handle arbitrary neural network (e.g., tensorflow-lite), you are supposed to write a new tensor-filter subplugin, not a custom subplugin. A custom subplugin is the method to embed a simple "function" into a pipeline. Such tensor-filter subplugins are here: https://github.com/nnstreamer/nnstreamer/tree/main/ext/nnstreamer/tensor_filter

Note that "custom-filter" is yet another "tensor-filter subplugin", which is a subplugin that accepts a C struct (or a C function) as a "model file".


> Location of Function Execution Code:

If you really mean "custom-filter", not "tensor-filter subplugin", it's executed at https://github.com/nnstreamer/nnstreamer/blob/4ad0b2729845ea9935e4a0f557f676f254859454/gst/nnstreamer/tensor_filter/tensor_filter_custom.c#L178-L187


> Verification of Data Type Transformation:
> When converting camera input data to tensor format through tensor transform, how can I verify that the data type has been correctly transformed?

Assuming that you are talking about tensor type conversion (typecasting), not video format conversion (videoconvert)...

You may do a filesink and verify the data with the output file. Many unit test cases include filesink, whose output files are compared with the golden data.

I don't think we have a filter that prints out tensor data to the console (normally, it would be too large to print out and look at it.)


> Issue with Model Invocation:

If you are really writing a custom filter, refer to the above code.
If you are actually writing a tensor-filter subplugin (e.g., https://github.com/nnstreamer/nnstreamer/tree/main/ext/nnstreamer/tensor_filter ), the invoke function is called at 
https://github.com/nnstreamer/nnstreamer/blob/4ad0b2729845ea9935e4a0f557f676f254859454/gst/nnstreamer/tensor_filter/tensor_filter.c#L1151-L1152

**Comment by myungjoo** [comment]
Without further follow-up questions, I'm closing this.
